test/block/blockupload.test.js

```diff
@@ -0,0 +1,805 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-env mocha */
+
+'use strict';
+
+const assert = require('assert');
+const fs = require('fs').promises;
+const nock = require('nock');
+const Path = require('path');
+const { BlockUpload } = require('../../lib/block/blockupload');
+
+const debug = require('debug');
+debug.enable('httptransfer*');
+
+describe('Block Upload', function () {
+    afterEach(async function () {
+        assert.ok(nock.isDone(), nock.pendingMocks(), 'check if all nocks have been used');
+        nock.cleanAll();
+    });
+
+    it('Block upload smoke test', async function () {
+        console.log("block upload test");
+
+        const HOST = 'http://test-aem-upload-201';
+        const testFile = Path.join(__dirname, 'file-1.jpg');
+        await fs.writeFile(testFile, 'hello world 123', 'utf8');
+
+        nock(HOST, {
+            reqheaders: {
+                'content-length': 15,
+                'content-type': 'image/jpeg'
+            }
+        })
+            .put('/path/to/file-1.jpg', 'hello world 123')
+            .reply(201);
+
+        const blockUpload = new BlockUpload();
+        const events = {
+            filestart: [],
+            fileprogress: [],
+            fileend: []
+        };
+        blockUpload.on('transferPart', (data) => {
+            events.filestart.push(data);
+        });
+        blockUpload.on('fileprogress', (data) => {
+            events.fileprogress.push(data);
+        });
+        blockUpload.on('aftertransfer', (data) => {
+            events.fileend.push(data);
+        });
+        const targetUrl = 'http://test-aem-upload-201/path/to/file-1.jpg';
+
+        await blockUpload.uploadFiles({
+            uploadFiles: [{
+                fileUrl: targetUrl,
+                filePath: testFile,
+                fileSize: 15
+            }],
```

Comment:
Nui currently does not use it, but maybe add a test with multiple files in the array?

Reply:
note: currently the exported function `uploadFileConcurrently` does not support multiple files: https://github.com/adobe/node-httptransfer/blob/feature/nui-1644/lib/file.js#L190
this means we would need to do a breaking change or export a new function to support this in the future. I think we should make `uploadFileConcurrently` -> `uploadFilesConcurrently` to avoid doing a another breaking change in the future... wdyt? @tmathern 
(also `uploadMultiPartFileConcurrently` -> `uploadMultiPartFilesConcurrently`)

Reply:
@jdelbick We can do the changes (it will be a breaking release anyway), but we need to add the tests to verify the functionality too.

Reply:
Added this